src/components/exercise9/fetching.jsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+import UserComponent from "./userComponent";
+
+export default function Fetch() {
+  const [user, setUser] = useState([]);
+  const [random, setRandom] = useState(0);
+
+  useEffect(
+    function () {
+      fetch("https://random-data-api.com/api/users/random_user?size=10")
+        .then((res) => res.json())
+        .then((data) => setUser(data));
+      console.log(user);
+    },
+    [random]
+  );
+
+  let data = user.map((item) => (
+    <UserComponent
+      key={item.id}
+      firstName={item.first_name}
+      lastName={item.last_name}
+      avatar={item.avatar}
+      employment={item.employment.title}
+      skill={item.employment.key_skill}
+      userName={item.username}
+      email={item.email}
+      tel={item.phone_number}
+      gender={item.gender}
+    />
+  ));
+
+  return (
+    <div className="flex flex-col items-center p-10">
+      <button
+        className="p-2 m-3 bg-red-500 rounded-lg text-white text-xl font-medium hover:bg-red-600 w-fit"
+        onClick={() => setRandom((prevRandom) => prevRandom + 1)}
+      >
```

Comment:
- I would suggest that you avoid updating the state using inline event handlers: While it is possible to define event handlers inline, it is generally considered bad practice because it can make your code harder to read and maintain. Instead, it is better to define your event handlers as methods on your function component. For more and clear explanations kindly check this [Handling events in React: Best practices and common mistakes](https://www.linkedin.com/pulse/handling-events-react-best-practices-common-mistakes-faizan/)

Reply:
Thank you